pylint_odoo/checkers/no_modules.py

```diff
@@ -408,6 +414,11 @@ def visit_functiondef(self, node):
                 self.add_message('method-required-super',
                                  node=node, args=(node.name))
 
+        if self.linter.is_message_enabled('old-api7-method-defined'):
+            first_args = [arg.name for arg in node.args.args][:3]
+            if first_args == ['self', 'cr', 'uid']:
```

Comment:
> Methods outside a class don't have `self`

Technically is not a `method` It is a `function`
The focus of this check are just `methods` (inside a class)

A good practice of pylint is create separated checks for methods and functions
We could create a new one with `old-api7-function-defined`
NOTE: I saw the hook and migration scripts using `cr` (without `uid`), but I don't saw functions using `cr, uid`, Do you have a example to create the new check?

> You should also look for `cursor` and `user`

Good point!

> I think we can check also if there are instructions like self.pool.

Good point too to detect use of old api, but Its is offtopic for this check, we could create `use-deprecated-attribute`